cmd/utils.go

```diff
@@ -229,3 +230,25 @@ func toS3ETag(etag string) string {
 
 	return etag
 }
+
+// Returns "/bucketName/objectName" for path-style or virtual-host-style requests.
+func getResource(path string, host string, domain string) (string, error) {
+	if domain == "" {
+		return path, nil
+	}
+
+	// If virtual-host-style is enabled construct the "resource" properly.
+	if strings.Contains(host, ":") {
+		// In bucket.mydomain.com:9000, strip out :9000
+		var err error
+		if host, _, err = net.SplitHostPort(host); err != nil {
```

Comment:
why not use `extractHostPort()` helper? you don't need to use `strings.Contains()` function here it would further simplify this.

Reply:
extractHostPort gives error if host is `localhost` i.e without port/scheme